src/components/ebay-calendar/component.js

```diff
@@ -0,0 +1,401 @@
+ *   rangeEnd: DayISO | undefined,
+ *   baseISO: DayISO,
+ *   disableBefore: DayISO | undefined,
+ *   disableAfter: DayISO | undefined,
+ *   disableWeekdays: number[],
+ *   disableList: DayISO[],
+ * }} State
+ * @extends {Marko.Component<Input, State>}
+ */
+export default class extends Marko.Component {
+    /**
+     * @param {Input} input
+     */
+    onCreate(input) {
+        const { firstDayOfWeek, weekdayLabels } = getWeekdayInfo(localeOverride(input.locale));
+        const todayISO = toISO(new Date());
+        this.state = {
+            focusISO: undefined,
+            baseISO: todayISO,
+            tabindexISO: todayISO,
+            todayISO,
+            offset: 0,
+            firstDayOfWeek,
+            weekdayLabels,
+            rangeStart: undefined,
+            rangeEnd: undefined,
+            disableBefore: undefined,
+            disableAfter: undefined,
+            disableWeekdays: [],
+            disableList: [],
+        };
+    }
+
+    /**
+     * @param {Input} input
+     */
+    onInput(input) {
+        if (input.selected) {
+            // If no selected times are visible, snap the view to the first one
+            const selectedISOs = Array.isArray(input.selected) ? input.selected : [input.selected];
+            const currFirstISO = this.getFirstVisibleISO();
+            const currLastISO = this.getLastVisibleISO(input);
+            const selectedTimeInView = selectedISOs.find(
+                (time) => time >= currFirstISO && time <= currLastISO
+            );
+            if (selectedTimeInView === undefined) {
+                this.state.baseISO = this.state.tabindexISO = selectedISOs[0];
+                this.state.offset = 0;
+            }
+        }
+        this.calculateRangeDisplay(input);
+
+        // handle changes regarding disabled dates
+        this.state.disableBefore = dateArgToISO(input.disableBefore);
+        this.state.disableAfter = dateArgToISO(input.disableAfter);
+        this.state.disableWeekdays = input.disableWeekdays ?? [];
+        this.state.disableList = /** @type {DayISO[]} */ (
+            input.disableList?.map(dateArgToISO) ?? []
+        );
+        if (this.isDisabled(this.state.tabindexISO)) {
+            this.state.tabindexISO = this.getFirstActiveISO(input);
+        }
+    }
+
+    /**
+     * @param {DayISO} iso
+     */
+    isDisabled(iso) {
+        return (
+            (this.state.disableBefore && iso < this.state.disableBefore) ||
+            (this.state.disableAfter && iso > this.state.disableAfter) ||
+            this.state.disableWeekdays.includes(fromISO(iso).getDay()) ||
+            this.state.disableList.includes(iso)
+        );
+    }
+
+    /**
+     * @param {DayISO} day
+     */
+    onDaySelect(day) {
+        this.emit('select', { iso: day });
+    }
+
+    /**
+     * @param {DayISO} day
+     */
+    onDayFocus(day) {
+        this.state.focusISO = this.state.tabindexISO = day;
+        this.calculateRangeDisplay();
```

Comment:
Might consider adding events emitted from here. 
Might be used for tracking. I'm not sure so I'm fine if we dont.